### ISSUE TITLE
Unknown reason Terminated

### ISSUE THREAD

**Issue by guoqiang0148666** [JoshuaPiinRueyPan/ViolenceDetection#16]
Hi, thank you for your sharing code~!

following your tutorial train on hockey dataset, l meet a Unknown reason Terminated.

any suggestions will be welcomed. 

the following is training log:

python3 Train.py

Using Network:  src.net.G2D19_P2OF_ResHB_1LSTM 

2019-03-04 22:10:58.491399: W tensorflow/core/graph/graph_constructor.cc:1265] Importing a graph with a lower producer version 24 into an existing graph with producer version 27. Shape inference will have run different parts of the graph with different producer versions.
In OpticalFlow:
	 pool2.shape =  (?, 112, 112, 32)
	 after transpose, out.shape =  (?, 112, 112, 32, 2)
	 before Conv2, out.shape =  (?, 112, 112, 64)
	 before Fc, out.shape =  (?, 28, 28, 64)
	 Fc final.shape =  (?, 1024)
opticalFlowOut.shape =  (?, 1024)
In CNN:
	 darknetOutput.shape =  (?, 7, 7, 1024)
	 after transpose, out.shape =  (?, 7, 7, 1024, 2)
	 before ConcatConv, out.shape =  (?, 7, 7, 2048)
	 before Fc, out.shape =  (?, 7, 7, 128)
	 Fc out.shape =  (?, 1024)
In Concat:
	 before Fc, out.shape =  (?, 2048)
before LSTM, shape =  (?, ?, 1024)

WARNING:tensorflow:From /media/guo/DataUbuntu/ViolenceDetection/src/Classifier.py:34: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See `tf.nn.softmax_cross_entropy_with_logits_v2`.

Reading Training set...
	 Done.

Reading Validation set...
	 Done.

Reading Test set...
	 Done.

2019-03-04 22:11:19.952292: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-03-04 22:11:20.146532: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1432] Found device 0 with properties: 
name: TITAN X (Pascal) major: 6 minor: 1 memoryClockRate(GHz): 1.531
pciBusID: 0000:01:00.0
totalMemory: 11.91GiB freeMemory: 11.12GiB
2019-03-04 22:11:20.477948: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1432] Found device 1 with properties: 
name: TITAN X (Pascal) major: 6 minor: 1 memoryClockRate(GHz): 1.531
pciBusID: 0000:02:00.0
totalMemory: 11.91GiB freeMemory: 11.75GiB
2019-03-04 22:11:20.481819: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1511] Adding visible gpu devices: 0, 1
2019-03-04 22:11:21.286405: I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-04 22:11:21.286446: I tensorflow/core/common_runtime/gpu/gpu_device.cc:988]      0 1 
2019-03-04 22:11:21.286455: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1001] 0:   N Y 
2019-03-04 22:11:21.286460: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1001] 1:   Y N 
2019-03-04 22:11:21.286937: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10761 MB memory) -> physical GPU (device: 0, name: TITAN X (Pascal), pci bus id: 0000:01:00.0, compute capability: 6.1)
2019-03-04 22:11:21.287347: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 11370 MB memory) -> physical GPU (device: 1, name: TITAN X (Pascal), pci bus id: 0000:02:00.0, compute capability: 6.1)
Path to save mode:  outputs/

Start Training...

Averaged GetBatch time:  0.0006592693328857422
Epoch: 1 ==================================================================================================================
	 Back Propergation time measurement:
		 duration:  244.98 s/epoch
		 average:  1.96 s/batch

	 validation:
	     loss: 0.4407      frame accuracy: 0.8793      best frame threshold: 22      video accuracy: 0.9200      duration: 59.85(s)

	 train:
	     loss: 0.4232      frame accuracy: 0.8943      given frame threshold: 22      video accuracy: 0.9160      duration: 284.97(s)

Terminated

environment information:
tensorflow 1.12.0
Cuda compilation tools, release 9.0, V9.0.176
cudnn 7.1.4

**Comment by JoshuaPiinRueyPan** [comment]
Hi,
   Could you give me more information about your hardware?  Moreover, could you make sure your Main Memory can handle the burden of loading the videos?  (In my computer, it will consume nearly 30GB of Main Memory!)
    Also, it seems that the whole process stuck in evaluating the Test Set.  Do you have any clue about this?  (e.g. do you add many or very large video as  your Test Set?).  Normally, the user have the issue of the loading thread may cause the training step failed.  However, you seems stuck in the evaluation...
